internal/bminventory/inventory.go

```diff
@@ -391,6 +404,20 @@ func (b *bareMetalInventory) updateImageInfoPostUpload(ctx context.Context, clus
 	return nil
 }
 
+func (b *bareMetalInventory) updateClusterProxyHashPostUpload(cluster *common.Cluster, clusterProxyHash string) error {
```

Comment:
I would merge this with updateImageInfoPostUpload() and have it update everything in the same transaction.  Also you need to set cluster.ProxyHash (or preferably cluster.ImageInfo.ProxyHash) so that it is returned to the user in this API call.